### PR TITLE
[Fix #8006] Fix infinite loop for `Style/IfUnlessModifier`

### LINKED ISSUE #7969: 0.83.0 broke this code with Style/IfUnlessModifier
## Expected behavior

The code should be fine, as it is with 0.82.0:
```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behavior

```
Offenses:

test.rb:6:17: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
    _customer = if params[:customer_id]
                ^^
test.rb:6:18: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
    _customer = (if params[:customer_id]
                 ^^
test.rb:6:72: C: [Corrected] Style/IfUnlessModifier: Modifier form of if makes the line too long.
    _customer = (CustomerCell.new(Customer.find(params[:customer_id])) if params[:customer_id])
                                                                       ^^

0 files inspected, 3 offenses detected, 3 offenses corrected
Infinite loop detected in /home/me/tmp/test.rb.
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:292:in `check_for_infinite_loop'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:275:in `block in iterate_until_no_changes'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `loop'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `iterate_until_no_changes'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:245:in `do_inspection_loop'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:124:in `block in file_offenses'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:148:in `file_offense_cache'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:122:in `file_offenses'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:113:in `process_file'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:92:in `block in each_inspected_file'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `reduce'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each_inspected_file'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:78:in `inspect_files'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/runner.rb:39:in `run'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli/command.rb:10:in `run'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli/environment.rb:17:in `run'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli.rb:65:in `run_command'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli.rb:72:in `execute_runners'
/home/me/.gem/gems/rubocop-0.83.0/lib/rubocop/cli.rb:41:in `run'
/home/me/.gem/gems/rubocop-0.83.0/exe/rubocop:13:in `block in <top (required)>'
/usr/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/home/me/.gem/gems/rubocop-0.83.0/exe/rubocop:12:in `<top (required)>'
/home/me/.gem/bin/rubocop:23:in `load'
/home/me/.gem/bin/rubocop:23:in `<main>'
```

## Steps to reproduce the problem

Create `test.rb` with this content:
```ruby
class X
  def y
    _customer = if params[:customer_id]
      CustomerCell.new(Customer.find(params[:customer_id]))
    end
  end
end
```
Create `.rubocop.yml` with this content:
```
Layout/LineLength:
  Max: 85
Layout/EndAlignment:
  EnforcedStyleAlignWith: variable
```

Run `rubocop -a test.rb`

## RuboCop version

Broken version:
```
$ rubocop -V
0.83.0 (using Parser 2.7.1.2, running on ruby 2.7.0 x86_64-linux-gnu)
```

Good version:
```
$ rubocop -V                   
0.82.0 (using Parser 2.7.1.2, running on ruby 2.7.0 x86_64-linux-gnu)
```

## Possibly related change

I haven't checked, but I assume this change could have broken this: https://github.com/rubocop-hq/rubocop/pull/7872

### LINKED ISSUE #8006: Infinite loop with Style/IfUnlessModifier
On Rubocop 0.84, with this statement:
```ruby
    timespan = if params[:timespan] && POSSIBLE_TIMESPANS.include?(params[:timespan])
                 params[:timespan].to_sym
               end
```
and LineLength set to 100, `rubocop -a` falls into infinite loop:
```
app/controllers/customer_dashboard_controller.rb:67:16: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
    timespan = if params[:timespan] && POSSIBLE_TIMESPANS.include?(params[:timespan])
               ^^
app/controllers/customer_dashboard_controller.rb:67:17: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
    timespan = (if params[:timespan] && POSSIBLE_TIMESPANS.include?(params[:timespan])
                ^^
app/controllers/customer_dashboard_controller.rb:67:42: C: [Corrected] Style/IfUnlessModifier: Modifier form of if makes the line too long.
    timespan = (params[:timespan].to_sym if params[:timespan] && POSSIBLE_TIMESPANS.include?(params[:timespan]))
                                         ^^

Infinite loop detected in .../app/controllers/customer_dashboard_controller.rb.
```

(Yeah, I know code itself is horrible! Already rewritten it by hand, but still)

### LINKED ISSUE #8025: Infinite loop with Style/IfUnlessModifier
*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------
**Note: This may be the same bug as #8006**
## Expected behavior

Either autocorrect or ignore this bit of code:
```ruby
      merge = if ver.respond_to?(:merge_source_id)
                get_version_merge_link(obj, ver)
              end
```
## Actual behavior

>$ rubocop -f html -o rubocop.html --only Style/IfUnlessModifier -a
Infinite loop detected in /vagrant/mushroom-observer/app/helpers/version_helper.rb.
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:292:in `check_for_infinite_loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:275:in `block in iterate_until_no_changes'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `iterate_until_no_changes'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:245:in `do_inspection_loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:124:in `block in file_offenses'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:148:in `file_offense_cache'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:122:in `file_offenses'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:113:in `process_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:92:in `block in each_inspected_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `reduce'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each_inspected_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:78:in `inspect_files'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:39:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command.rb:10:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/environment.rb:17:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:65:in `run_command'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:72:in `execute_runners'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:41:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/exe/rubocop:13:in `block in <top (required)>'
/home/vagrant/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/benchmark.rb:308:in `realtime'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/exe/rubocop:12:in `<top (required)>'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/rubocop:23:in `load'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/rubocop:23:in `<main>'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/ruby_executable_hooks:24:in `eval'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/ruby_executable_hooks:24:in `<main>'
VM:/vagrant/mushroom-observer 
$ rubocop -f html -o rubocop.html --only Style/IfUnlessModifier -a app/helpers/version_helper.rb
Infinite loop detected in /vagrant/mushroom-observer/app/helpers/version_helper.rb.
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:292:in `check_for_infinite_loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:275:in `block in iterate_until_no_changes'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `iterate_until_no_changes'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:245:in `do_inspection_loop'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:124:in `block in file_offenses'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:148:in `file_offense_cache'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:122:in `file_offenses'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:113:in `process_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:92:in `block in each_inspected_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `reduce'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each_inspected_file'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:78:in `inspect_files'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/runner.rb:39:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/command.rb:10:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli/environment.rb:17:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:65:in `run_command'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:72:in `execute_runners'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/lib/rubocop/cli.rb:41:in `run'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/exe/rubocop:13:in `block in <top (required)>'
/home/vagrant/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/benchmark.rb:308:in `realtime'
/home/vagrant/.rvm/gems/ruby-2.6.6/gems/rubocop-0.83.0/exe/rubocop:12:in `<top (required)>'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/rubocop:23:in `load'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/rubocop:23:in `<main>'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/ruby_executable_hooks:24:in `eval'
/home/vagrant/.rvm/gems/ruby-2.6.6/bin/ruby_executable_hooks:24:in `<main>'

## Steps to reproduce the problem
The problem can be reproduced by running
```
$ rubocop -f html -o rubocop.html --only Style/IfUnlessModifier -a
```
on a file containing
```ruby
      merge = if ver.respond_to?(:merge_source_id)
                get_version_merge_link(obj, ver)
              end
```
## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ rubocop -V
0.83.0 (using Parser 2.7.1.2, running on ruby 2.6.6 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #8006 
Closes #8025 
Closes #7969

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/mixin/statement_modifier.rb [hunk 1]
```diff
@@ -42,8 +42,9 @@ def modifier_fits_on_single_line?(node)
       def length_in_modifier_form(node, cond)
         keyword = node.loc.keyword
         indentation = keyword.source_line[/^\s*/]
-        line_length("#{indentation}#{node.body.source} #{keyword.source} " \
-                    "#{cond.source}")
+        source = "#{indentation}#{node.body.source} #{keyword.source} #{cond.source}"
+        source = "#{node.parent.loc.name} = #{source}" if node.parent&.lvasgn_type?
```
Comment (marcandre) [inline lib/rubocop/cop/mixin/statement_modifier.rb:46]: Best use an `else` and build `source` only once

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 * [#8008](https://github.com/rubocop-hq/rubocop/issues/8008): Fix an error for `Lint/SuppressedException` when empty rescue block in `def`. ([@koic][])
 * [#8012](https://github.com/rubocop-hq/rubocop/issues/8012): Fix an incorrect autocorrect for `Lint/DeprecatedOpenSSLConstant` when deprecated OpenSSL constant is used in a block. ([@koic][])
+* [#8006](https://github.com/rubocop-hq/rubocop/issues/8006): Fix infinite loop for `Style/IfUnlessModifier`. ([@tejasbubane][])
```
Comment (mwolman) [inline CHANGELOG.md:9]: And issue 8025? 🤔

File: spec/rubocop/cop/style/if_unless_modifier_spec.rb [hunk 3]
```diff
@@ -421,6 +421,18 @@ def f
     expect(corrected).to eq "a = (1 if b)\n"
   end
 
+  context 'assignment pushes inline length over limit' do
```
Comment (mwolman) [inline spec/rubocop/cop/style/if_unless_modifier_spec.rb:424]:
```suggestion
  context 'when the assignment pushes inline length over limit' do
```

File: CHANGELOG.md (hunk 2 above)
Comment (koic) [reply CHANGELOG.md:9]: 8006 is fine. Both have the same issue.

File: lib/rubocop/cop/mixin/statement_modifier.rb (hunk 1 above)
Comment (tejasbubane) [reply lib/rubocop/cop/mixin/statement_modifier.rb]: @marcandre Done. Had to pull out `condition_source` to overcome `Metrics/AbcSize` offense.

File: spec/rubocop/cop/style/if_unless_modifier_spec.rb (hunk 3 above)
Comment (tejasbubane) [reply spec/rubocop/cop/style/if_unless_modifier_spec.rb]: Fixed.

**Comment by joe-p** [comment]
The following line will still produce an infinite loop with this commit pulled into master:

```ruby
puts "                                                                                                    #{' ' if true}"
```

```
$ rubocop -ad test.rb
For /home/joe/rubo_test/test: Default configuration from /home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/config/default.yml
Inspecting 1 file
Scanning /home/joe/rubo_test/test/test.rb
C

Offenses:

test.rb:3:109: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
puts "                                                                                                    #{if true
                                                                                                            ^^
test.rb:3:113: C: [Corrected] Style/IfUnlessModifier: Modifier form of if makes the line too long.
puts "                                                                                                    #{' ' if true}"
                                                                                                                ^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /home/joe/rubo_test/test/test.rb.
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:290:in `check_for_infinite_loop'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:273:in `block in iterate_until_no_changes'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:272:in `loop'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:272:in `iterate_until_no_changes'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:243:in `do_inspection_loop'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:122:in `block in file_offenses'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:146:in `file_offense_cache'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:120:in `file_offenses'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:111:in `process_file'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:90:in `block in each_inspected_file'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:89:in `each'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:89:in `reduce'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:89:in `each_inspected_file'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:78:in `inspect_files'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/runner.rb:39:in `run'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli/command.rb:10:in `run'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli/environment.rb:17:in `run'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli.rb:65:in `run_command'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli.rb:72:in `execute_runners'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/lib/rubocop/cli.rb:41:in `run'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/exe/rubocop:13:in `block in <top (required)>'
/home/joe/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/gems/rubocop-0.84.0/exe/rubocop:12:in `<top (required)>'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/bin/rubocop:23:in `load'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/bin/rubocop:23:in `<main>'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/bin/ruby_executable_hooks:24:in `eval'
/home/joe/.rvm/gems/ruby-2.7.1@kisscraft/bin/ruby_executable_hooks:24:in `<main>'
Finished in 0.10348649999991721 seconds
```

File: lib/rubocop/cop/mixin/statement_modifier.rb (hunk 1 above)
Comment (marcandre) [reply lib/rubocop/cop/mixin/statement_modifier.rb]: Looks great to me 👍

**Comment by tejasbubane** [comment]
@joe-p Now it should not.

File: spec/rubocop/cop/style/if_unless_modifier_spec.rb [hunk 4]
```diff
@@ -421,6 +421,26 @@ def f
...
+
+    it 'does not result in infinite loop with assignment' do
+      expect_no_offenses(<<~RUBY)
+        timespan = if params[:timespan] && POSSIBLE_TIMESPANS.include?(params[:timespan])
+                     params[:timespan].to_sym
+                   end
+      RUBY
+    end
+
+    it 'does not result in infinite loop with any other statement' do
```
Comment (koic) [inline spec/rubocop/cop/style/if_unless_modifier_spec.rb:435]: An infinite loop error is caused by a combination of cops. Con you add a reproduction test for infinite loop error to spec/rubocop/cli/cli_autocorrect_spec.rb? E.g.: https://github.com/rubocop-hq/rubocop/pull/7901/files#diff-0e7dbdd6e99e0defc2bab2cb544b42da
And I think the description of this test better to be updated to not register an offense.

**Comment by iGEL** [comment]
I'd love to see this merged. Is there anything I could help with?

**Comment by tejasbubane** [comment]
@iGEL I missed @koic's comment. Working on it now.

**Comment by marcandre** [comment]
I also opened #8283 (but didn't know about this)

As I wrote there, I feel it is imperative that the code that builds the replacement string should be used for the calculation length.

The same bugs exist for `WhileUntilModifier` too.

The refactoring I made yesterday should help a bit.

**Comment by tejasbubane** [comment]
> I feel it is imperative that the code that builds the replacement string should be used for the calculation length.

I agree with @marcandre's approach. Closing this for now.

**Pull request closed without merging**